UnityProject/Assets/Scripts/Objects/Other/PineTreeXmas.cs

```diff
@@ -19,47 +19,55 @@ public class PineTreeXmas : NetworkBehaviour, ICheckedInteractable<HandApply>
 		[SerializeField] private GameObject giftObject;
 		[SerializeField] private AddressableAudioSource ambientReminder;
 
-		[SyncVar] private List<string> giftedPlayers = new List<string>();
+		private List<string> giftedPlayers = new List<string>();
 		private bool canPickUpGifts;
 
 		private SpriteHandler spriteHandler;
 
-		private void Awake()
+		private void Start()
 		{
 			spriteHandler = GetComponentInChildren<SpriteHandler>();
 			if (TimedEventsManager.Instance.ActiveEvents.Contains(eventData))
 			{
 				canPickUpGifts = true;
 				spriteHandler.SetSpriteSO(xmasSpriteSO);
-				PlayEasterEgg();
+				UpdateManager.Add(PlayEasterEgg, 60);
 			}
 		}
 
-		private async void PlayEasterEgg()
+		[ContextMenu("Play the reminder of our finite time")]
+		private void PlayEasterEgg()
 		{
-			await Task.Delay(60 * 1000); // millaseconds
+			if(gameObject == null) return;
+
 			if (DMMath.Prob(10))
 			{
 				_ = SoundManager.PlayNetworkedAtPosAsync(ambientReminder, gameObject.WorldPosServer());
 			}
-			PlayEasterEgg();
+		}
+
+		[Command]
+		private bool CmdIsGifted(string userID)
```

Comment:
Cant do returns

Reply:
how would i do it then?

Reply:
tbh, i'd just have a bool in WillInteract which'll be set true on the first time the client clicks, which will then block subsequent clicks. Keep the list serverside and check that server side in WillInteract to prevent players who quit and rejoin to get another present. Dont need to bother with commands and RPC's

Reply:
is it done?